### ISSUE TITLE
useContext returns undefined

### ISSUE THREAD

**Issue by hazae41** [alephjs/aleph.js#320]
Hey, since the latest alpha 33, `useContext()` returns undefined even if the context is defined

```typescript
<AppContext.Provider value={...}>
  <MyComponent />
</AppContext.Provider>

...

function MyComponent(){
  const app = useContext(AppContext)
  console.log("app", app) // undefined

  ...
}
```

My imports map 

```json
{
    "aleph/": "https://deno.land/x/aleph@v0.3.0-alpha.33/",
    "framework": "https://deno.land/x/aleph@v0.3.0-alpha.33/framework/core/mod.ts",
    "framework/react": "https://deno.land/x/aleph@v0.3.0-alpha.33/framework/react/mod.ts",
    "react": "https://esm.sh/react@17.0.2",
    "react-dom": "https://esm.sh/react-dom@17.0.2",
}
```

**Comment by hazae41** [comment]
It does work when using `aleph build`
It seems to only happen using `aleph dev`

**Comment by ije** [comment]
@hazae41 I have no idea...can you please post the repo?

**Comment by hazae41** [comment]
@ije 

Here is an example of the bug,

It seems to only happen in certain conditions, when the context, the provider, and the consumer are in different files

Working: https://github.com/hazae41/create-aleph-vercel-app/tree/master/app
Not working: https://github.com/hazae41/create-aleph-vercel-app/tree/bug/app

**Comment by hazae41** [comment]
I just moved Context from `libs/context.ts` to `comps/context.tsx` and it works

**Comment by hazae41** [comment]
After the move, I added loggers

One in the consume: does nothing, still works

```typescript
import { Context } from "./context.tsx";

export const Hello = () => {
  console.log("Context in consumer", Context)

  const API = useContext(Context)
  ...
}
```

One in provider, the bug happens if I add the logger

```typescript
export const Context =
  createContext<string | undefined>(undefined)

export const ContextProvider = (props: {
  children: ReactNode
}) => {
  console.log("Context in provider", Context)

  ...
}
```

This is very weird

**Comment by FallingSnow** [comment]
Even `aleph init hello && cd hello && aleph run dev` doesn't work.

```
Error: The requested module '../../../../../../esm.sh/react@8.10.0.js' does not provide an export named 'useContext'
TypeError: The requested module '../../../../../../esm.sh/react@8.10.0.js' does not provide an export named 'useContext'
    at async Application.importModule (https://deno.land/x/aleph@v0.3.0-alpha.33/server/app.ts:837:12)
    at async Renderer.renderPage (https://deno.land/x/aleph@v0.3.0-alpha.33/server/ssr.ts:113:42)
    at async https://deno.land/x/aleph@v0.3.0-alpha.33/server/app.ts:523:26
    at async Renderer.useCache (https://deno.land/x/aleph@v0.3.0-alpha.33/server/ssr.ts:89:24)
    at async Application.getPageHTML (https://deno.land/x/aleph@v0.3.0-alpha.33/server/app.ts:522:20)
    at async Server.handle (https://deno.land/x/aleph@v0.3.0-alpha.33/server/server.ts:204:30)
```

**Comment by hazae41** [comment]
> Even `aleph init hello && cd hello && aleph run dev` doesn't work.
> 
> 
> 
> ```
> 
> Error: The requested module '../../../../../../esm.sh/react@8.10.0.js' does not provide an export named 'useContext'
> 
> TypeError: The requested module '../../../../../../esm.sh/react@8.10.0.js' does not provide an export named 'useContext'
> 
>     at async Application.importModule (https://deno.land/x/aleph@v0.3.0-alpha.33/server/app.ts:837:12)
> 
>     at async Renderer.renderPage (https://deno.land/x/aleph@v0.3.0-alpha.33/server/ssr.ts:113:42)
> 
>     at async https://deno.land/x/aleph@v0.3.0-alpha.33/server/app.ts:523:26
> 
>     at async Renderer.useCache (https://deno.land/x/aleph@v0.3.0-alpha.33/server/ssr.ts:89:24)
> 
>     at async Application.getPageHTML (https://deno.land/x/aleph@v0.3.0-alpha.33/server/app.ts:522:20)
> 
>     at async Server.handle (https://deno.land/x/aleph@v0.3.0-alpha.33/server/server.ts:204:30)
> 
> ```

This is not the same bug

Mine builds but the context is undefined, yours doesn't build

**Comment by ije** [comment]
fixed in bad514f0546d57556e82bebc457245aef7a68cac, this is casued by the **HMR**: when you change your code the client hmr will re-import the changed module using a different URL(hash) that may make this problem because the `Context.Provider` and `Context.Consumer` is not a same object at all!
